desc/objectives/_generic.py

```diff
@@ -15,6 +19,262 @@
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Has no effect for this objective.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    name : str, optional
+        Name of the objective function.
+
+    # TODO: add example
+
+    """
+
+    _units = "(Unknown)"
+    _print_value_fmt = "External objective value: {:10.3e}"
+
+    def __init__(
+        self,
+        eq,
+        fun,
+        dim_f,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=False,
+        normalize_target=False,
+        loss_function=None,
+        fd_step=1e-4,  # TODO: generalize this to allow a vector of different scales
+        vectorized=False,  # False or int
+        name="external",
+        **kwargs,
+    ):
+        assert isinstance(vectorized, bool) or isinstance(vectorized, int)
+        if target is None and bounds is None:
+            target = 0
+        self._eq = eq.copy()
+        self._fun = fun
+        self._dim_f = dim_f
+        self._fd_step = fd_step
+        self._vectorized = vectorized
+        self._kwargs = kwargs
+        if self._vectorized:
+            try:  # spawn a new environment so the backend can be set to numpy
+                mp.set_start_method("spawn")
+            except RuntimeError:  # context can only be set once
+                pass
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode="fwd",
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        self._scalar = self._dim_f == 1
+        self._constants = {"quad_weights": 1.0}
+
+        if self._vectorized:
+            max_processes = (
+                self._vectorized
+                if isinstance(self._vectorized, int)
+                else os.cpu_count()
+            )
+            self._pool = mp.Pool(processes=max_processes)
```

Comment:
so this isn't used?

Reply:
Correct, I forgot to remove it after testing